cosmo_tester/test_suites/service_composition/shared_resource_test.py

```diff
@@ -20,3 +24,59 @@ def test_shared_resource(image_based_manager, ssh_key, logger, test_config):
         _verify_deployments_and_nodes(app, 2)
         _check_custom_execute_operation(app, logger)
         _verify_custom_execution_cancel_and_resume(app, logger)
+
+
+@pytest.fixture(scope='function')
+def managers(request, ssh_key, module_tmpdir, test_config, logger):
+    hosts = Hosts(ssh_key, module_tmpdir, test_config, logger, request, 2)
```

Comment:
I think you need the 'passed' logic here (like the ones in the multi-tenant test), to make the conditional teardown work properly.